### PR TITLE
[New Extension] Switch Game Play History

### PR DESCRIPTION
**Pull request by ghost**
## Description

Browse your Nintendo Switch gameplay history. Including information such as first played time, last played time, and total playtime for each game.

## Screencast

switch-game-play-history-1
image

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Could you make a video of the flow and commands? Does this work outside japan?

**Comment by kvdo2** [comment]
> Could you make a video of the flow and commands? Does this work outside japan?

According to incomplete statistics, only the data of Hong Kong regional accounts is not supported at present. I will provide the video later.

**Comment by kvdo2** [comment]
I wrote down the detailed usage in [Readme](https://github.com/kvdo2/extensions/tree/switch-game-play-history/extensions/switch-game-play-history).

This is a video demonstration.

https://github.com/raycast/extensions/assets/78251524/2311166e-21d2-4bb7-bb9e-746b65a66ea9

File: extensions/switch-game-play-history/src/components/DailyPlayHistoriesSection.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+            value: [
+              "\u200B",
+              "\u3000".repeat(Math.floor((item.totalPlayedMinutes / timeBarMaxValue) * 12)),
+              item.totalPlayedMinutes + " min",
+              "\u200B",
+            ].join(""),
+            color:
+              item.totalPlayedMinutes < 60 ? Color.Green : item.totalPlayedMinutes < 120 ? Color.Orange : Color.Red,
+          },
+        };
```
Comment (pernielsentikaer) [inline extensions/switch-game-play-history/src/components/DailyPlayHistoriesSection.tsx:29]: This is not the way to use tags but it's creative 🙂 

You could also use [getProgressIcon](https://developers.raycast.com/utilities/icons/getprogressicon) with a tooltip instead 🙂

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @kvdo2 👋 

Could you look into this

File: extensions/switch-game-play-history/src/components/DailyPlayHistoriesSection.tsx (hunk 1 above)
Comment (kvdo2) [reply extensions/switch-game-play-history/src/components/DailyPlayHistoriesSection.tsx:29]: Do I have to change it? If this way is not allowed, then I can only cancel the display of the progress bar. GetProgressIcon only has the style of Ring, which doesn't quite meet my needs.

Comment (pernielsentikaer) [reply extensions/switch-game-play-history/src/components/DailyPlayHistoriesSection.tsx:29]: Yeah, that's unfortunately at the moment not allowed, so it needs to be just a tag without the text length 🙂

**Review by pernielsentikaer — APPROVED** [review]
Hi 👋 

Looks good to me, approved 🔥

**Pull request merged**